### PR TITLE
Doc: clarify TimePeriod/ScheduledDowntime time zone handling

### LINKED ISSUE #7069: Documentation should be more clear about time zones
The [v2.10.4 (current as of this writing) documentation about TimePeriod objects](https://github.com/Icinga/icinga2/blob/v2.10.4/doc/09-object-types.md#timeperiod-) does not explicitly details the "ranges" format and this issue is specifically about the lack of details about time zones.

While I hope that TimePeriods will one day support time zones (see #5374), it would be good if the documentation could be more clear about time zones handling:

- tell the reader that the "ranges" are expressed in local time, as handled by the operating system standard libraries
- tell the reader when time ranges are evaluated and if / how they account for time zone changes

Example scenario:

- I have a time range that says 07:30 to 00:00
- a little before midnight Icinga2 runs the last check of the day, the next one should be after 07:30 _local time_
- but between 00:00 and 07:30 we get a time zone change (DST starts / ends)
- does the interaction between the `TimePeriod` and the check scheduler and other parts work as expected and will I get the next check run around 07:30 _on the "new" localtime_, or maybe Icinga2 scheduled it with the old localtime and it will get run one hour earlier / one hour later?

### PR DESCRIPTION
**Pull request by Al2Klimov**
fixes #7069

### CODE REVIEW

**Review by julianbrost — COMMENTED** [review]
The change itself is fine but I wouldn't say that this alone fixes #7069. For example, it does not explain what exactly happens, if a DST change happens within an interval. Other questions that might be interesting and I can't answer from the docs: what happens if the start or end is something like 2:30 on a DST change day and this time either exists twice or not at all?

**Comment by Al2Klimov** [comment]
In you opinion: Shall we really promise anything e.g. regarding start/end at the twice appearing (i.e. ambiguous) 2020-10-25T02:30:00?

**Comment by Al2Klimov** [comment]
Or – even better – at 2020-03-29T02:30:00 which doesn't exist at all?

**Comment by julianbrost** [comment]
Well, something has to happen if you specify this in the config. I don't think it's a good idea to leave it as a surprise what happens exactly. [The docs even use 2:00-3:00 in examples](https://icinga.com/docs/icinga-2/latest/doc/08-advanced-topics/#recurring-downtimes), what will happen on days with DST changes?

**Comment by Al2Klimov** [comment]
Do you think the surprise is a bad idea due to our examples and the fact that we don’t advise against such or in general?

**Comment by julianbrost** [comment]
I'm fine with this PR, but it should not have "fixes #7069" as I think we can do better than this.

What we should do in my opinion to actually fix that issue is to take a look at the current behavior including all kinds of nasty edge cases and depending on whether it makes sense, document it as is or fix the implementation and then document it.

**Comment by Al2Klimov** [comment]
## CEST -> CET: CEST

```
➜  icinga2 git:(master) cat 8001/etc/icinga2/conf.d/8001.conf
apply ScheduledDowntime "8001" to Host {
	author = "me"
	comment = " "
	ranges = { sunday = "2:15-2:45" }
	assign where true
}
➜  icinga2 git:(master) docker run --rm -itv $(pwd)/8001:/data -e LD_PRELOAD=/data/libfaketime/src/libfaketime.so.1 -e FAKETIME=-82d -e TZ=Europe/Berlin icinga/icinga2:master date
[2021-01-14 12:39:42 +0100] information/DockerEntrypoint: Initializing /data as we're the init process (PID 1)
[2021-01-14 12:39:42 +0100] information/DockerEntrypoint: Checking "/data/etc/icinga2"
[2021-01-14 12:39:42 +0100] information/DockerEntrypoint: Checking "/data/var/cache/icinga2"
[2021-01-14 12:39:42 +0100] information/DockerEntrypoint: Checking "/data/var/lib/icinga2"
[2021-01-14 12:39:42 +0100] information/DockerEntrypoint: Checking "/data/var/log/icinga2"
[2021-01-14 12:39:42 +0100] information/DockerEntrypoint: Checking "/data/var/run/icinga2"
[2021-01-14 12:39:42 +0100] information/DockerEntrypoint: Checking "/data/var/spool/icinga2"
[2021-01-14 12:39:42 +0100] information/DockerEntrypoint: Checking "/var/lib/icinga2/certs/ca.crt"
[2021-01-14 12:39:42 +0100] information/DockerEntrypoint: Looking up "dumb-init" in $PATH
[2021-01-14 12:39:42 +0100] information/DockerEntrypoint: Running "/usr/bin/dumb-init"
Sat Oct 24 13:39:42 CEST 2020
➜  icinga2 git:(master) docker run --rm -itv $(pwd)/8001:/data -e LD_PRELOAD=/data/libfaketime/src/libfaketime.so.1 -e FAKETIME=-82d -e TZ=Europe/Berlin icinga/icinga2:master
[2021-01-14 12:39:55 +0100] information/DockerEntrypoint: Initializing /data as we're the init process (PID 1)
[2021-01-14 12:39:55 +0100] information/DockerEntrypoint: Checking "/data/etc/icinga2"
[2021-01-14 12:39:55 +0100] information/DockerEntrypoint: Checking "/data/var/cache/icinga2"
[2021-01-14 12:39:55 +0100] information/DockerEntrypoint: Checking "/data/var/lib/icinga2"
[2021-01-14 12:39:55 +0100] information/DockerEntrypoint: Checking "/data/var/log/icinga2"
[2021-01-14 12:39:55 +0100] information/DockerEntrypoint: Checking "/data/var/run/icinga2"
[2021-01-14 12:39:55 +0100] information/DockerEntrypoint: Checking "/data/var/spool/icinga2"
[2021-01-14 12:39:55 +0100] information/DockerEntrypoint: Checking "/var/lib/icinga2/certs/ca.crt"
[2021-01-14 12:39:55 +0100] information/DockerEntrypoint: Looking up "dumb-init" in $PATH
[2021-01-14 12:39:55 +0100] information/DockerEntrypoint: Running "/usr/bin/dumb-init"
[2020-10-24 13:39:55 +0200] information/cli: Icinga application loader (version: r2.12.0-1)
[2020-10-24 13:39:55 +0200] information/cli: Loading configuration file(s).
[2020-10-24 13:39:55 +0200] information/ConfigItem: Committing config item(s).
[2020-10-24 13:39:55 +0200] information/ConfigItem: Instantiated 1 NotificationComponent.
[2020-10-24 13:39:55 +0200] information/ConfigItem: Instantiated 1 CheckerComponent.
[2020-10-24 13:39:55 +0200] information/ConfigItem: Instantiated 1 User.
[2020-10-24 13:39:55 +0200] information/ConfigItem: Instantiated 1 UserGroup.
[2020-10-24 13:39:55 +0200] information/ConfigItem: Instantiated 3 ServiceGroups.
[2020-10-24 13:39:55 +0200] information/ConfigItem: Instantiated 3 TimePeriods.
[2020-10-24 13:39:55 +0200] information/ConfigItem: Instantiated 11 Services.
[2020-10-24 13:39:55 +0200] information/ConfigItem: Instantiated 3 Zones.
[2020-10-24 13:39:55 +0200] information/ConfigItem: Instantiated 2 ScheduledDowntimes.
[2020-10-24 13:39:55 +0200] information/ConfigItem: Instantiated 2 HostGroups.
[2020-10-24 13:39:55 +0200] information/ConfigItem: Instantiated 2 NotificationCommands.
[2020-10-24 13:39:55 +0200] information/ConfigItem: Instantiated 12 Notifications.
[2020-10-24 13:39:55 +0200] information/ConfigItem: Instantiated 1 IcingaApplication.
[2020-10-24 13:39:55 +0200] information/ConfigItem: Instantiated 1 Host.
[2020-10-24 13:39:55 +0200] information/ConfigItem: Instantiated 1 Endpoint.
[2020-10-24 13:39:55 +0200] information/ConfigItem: Instantiated 235 CheckCommands.
[2020-10-24 13:39:55 +0200] information/ScriptGlobal: Dumping variables to file '/var/cache/icinga2/icinga2.vars'
[2020-10-24 13:39:55 +0200] information/ConfigObject: Restoring program state from file '/var/lib/icinga2/icinga2.state'
[2020-10-24 13:39:55 +0200] information/ConfigObject: Restored 283 objects. Loaded 28 new objects without state.
[2020-10-24 13:39:56 +0200] information/ConfigItem: Triggering Start signal for config items
[2020-10-24 13:39:56 +0200] information/NotificationComponent: 'notification' started.
[2020-10-24 13:39:56 +0200] information/CheckerComponent: 'checker' started.
[2020-10-24 13:39:56 +0200] information/ConfigItem: Activated all objects.
[2020-10-24 13:40:56 +0200] information/ConfigObjectUtility: Created and activated object 'daef7cb16335!load!b0ea7cf7-e899-4c77-a0d4-ef4c581ff5ad' of type 'Downtime'.
[2020-10-24 13:40:56 +0200] information/Downtime: Added downtime 'daef7cb16335!load!b0ea7cf7-e899-4c77-a0d4-ef4c581ff5ad' between '2020-10-25 02:00:00' and '2020-10-25 02:00:00', author: 'icingaadmin', fixed
[2020-10-24 13:40:56 +0200] information/ConfigObjectUtility: Created and activated object 'daef7cb16335!95e534ba-7f92-4bdb-896e-bd35d93b6440' of type 'Downtime'.
[2020-10-24 13:40:56 +0200] information/Downtime: Added downtime 'daef7cb16335!95e534ba-7f92-4bdb-896e-bd35d93b6440' between '2020-10-25 02:15:00' and '2020-10-25 02:45:00', author: 'me', fixed
^C[2021-01-14 12:41:08 +0100] information/Application: Received request to shut down.
[2021-01-14 12:41:08 +0100] information/Application: Received request to shut down.
[2020-10-24 13:41:08 +0200] information/Application: Shutting down...
[2020-10-24 13:41:08 +0200] information/CheckerComponent: 'checker' stopped.
[2020-10-24 13:41:08 +0200] information/NotificationComponent: 'notification' stopped.
[2020-10-24 13:41:08 +0200] information/ConfigObject: Dumping program state to file '/var/lib/icinga2/icinga2.state'
[2020-10-24 13:41:08 +0200] information/IcingaApplication: Icinga has shut down.
➜  icinga2 git:(master) find 8001 -name '*95e534ba-7f92-4bdb-896e-bd35d93b6440*'
8001/var/lib/icinga2/api/packages/_api/601a910f-4f1e-4911-b656-e85ea7787650/conf.d/downtimes/daef7cb16335!95e534ba-7f92-4bdb-896e-bd35d93b6440.conf
➜  icinga2 git:(master) cat 8001/var/lib/icinga2/api/packages/_api/601a910f-4f1e-4911-b656-e85ea7787650/conf.d/downtimes/daef7cb16335\!95e534ba-7f92-4bdb-896e-bd35d93b6440.conf
object Downtime "95e534ba-7f92-4bdb-896e-bd35d93b6440" ignore_on_error {
	author = "me"
	comment = " "
	config_owner = "daef7cb16335!8001"
	duration = 0.000000
	end_time = 1603586700.000000
	entry_time = 1603539656.069030
	fixed = true
	host_name = "daef7cb16335"
	scheduled_by = "daef7cb16335!8001"
	start_time = 1603584900.000000
	triggered_by = ""
	version = 1603539656.069074
}
➜  icinga2 git:(master) date -r 1603584900
So 25 Okt 2020 02:15:00 CEST
➜  icinga2 git:(master) date -r 1603586700
So 25 Okt 2020 02:45:00 CEST
➜  icinga2 git:(master)
```

## CET -> CEST: CET

```
➜  icinga2 git:(master) cat 8001/etc/icinga2/conf.d/8001.conf
apply ScheduledDowntime "8001" to Host {
	author = "me"
	comment = " "
	ranges = { sunday = "2:15-2:45" }
	assign where true
}
➜  icinga2 git:(master) docker run --rm -itv $(pwd)/8001:/data -e LD_PRELOAD=/data/libfaketime/src/libfaketime.so.1 -e FAKETIME=+72d -e TZ=Europe/Berlin icinga/icinga2:master date
[2021-01-14 13:01:15 +0100] information/DockerEntrypoint: Initializing /data as we're the init process (PID 1)
[2021-01-14 13:01:15 +0100] information/DockerEntrypoint: Checking "/data/etc/icinga2"
[2021-01-14 13:01:15 +0100] information/DockerEntrypoint: Checking "/data/var/cache/icinga2"
[2021-01-14 13:01:15 +0100] information/DockerEntrypoint: Checking "/data/var/lib/icinga2"
[2021-01-14 13:01:15 +0100] information/DockerEntrypoint: Checking "/data/var/log/icinga2"
[2021-01-14 13:01:15 +0100] information/DockerEntrypoint: Checking "/data/var/run/icinga2"
[2021-01-14 13:01:15 +0100] information/DockerEntrypoint: Checking "/data/var/spool/icinga2"
[2021-01-14 13:01:15 +0100] information/DockerEntrypoint: Checking "/var/lib/icinga2/certs/ca.crt"
[2021-01-14 13:01:15 +0100] information/DockerEntrypoint: Looking up "dumb-init" in $PATH
[2021-01-14 13:01:15 +0100] information/DockerEntrypoint: Running "/usr/bin/dumb-init"
Sat Mar 27 13:01:15 CET 2021
➜  icinga2 git:(master) docker run --rm -itv $(pwd)/8001:/data -e LD_PRELOAD=/data/libfaketime/src/libfaketime.so.1 -e FAKETIME=+72d -e TZ=Europe/Berlin icinga/icinga2:master
[2021-01-14 13:01:21 +0100] information/DockerEntrypoint: Initializing /data as we're the init process (PID 1)
[2021-01-14 13:01:21 +0100] information/DockerEntrypoint: Checking "/data/etc/icinga2"
[2021-01-14 13:01:21 +0100] information/DockerEntrypoint: Checking "/data/var/cache/icinga2"
[2021-01-14 13:01:21 +0100] information/DockerEntrypoint: Checking "/data/var/lib/icinga2"
[2021-01-14 13:01:21 +0100] information/DockerEntrypoint: Checking "/data/var/log/icinga2"
[2021-01-14 13:01:21 +0100] information/DockerEntrypoint: Checking "/data/var/run/icinga2"
[2021-01-14 13:01:21 +0100] information/DockerEntrypoint: Checking "/data/var/spool/icinga2"
[2021-01-14 13:01:21 +0100] information/DockerEntrypoint: Checking "/var/lib/icinga2/certs/ca.crt"
[2021-01-14 13:01:21 +0100] information/DockerEntrypoint: Looking up "dumb-init" in $PATH
[2021-01-14 13:01:21 +0100] information/DockerEntrypoint: Running "/usr/bin/dumb-init"
[2021-03-27 13:01:21 +0100] information/cli: Icinga application loader (version: r2.12.0-1)
[2021-03-27 13:01:21 +0100] information/cli: Loading configuration file(s).
[2021-03-27 13:01:21 +0100] information/ConfigItem: Committing config item(s).
[2021-03-27 13:01:22 +0100] information/ConfigItem: Instantiated 1 NotificationComponent.
[2021-03-27 13:01:22 +0100] information/ConfigItem: Instantiated 1 CheckerComponent.
[2021-03-27 13:01:22 +0100] information/ConfigItem: Instantiated 1 User.
[2021-03-27 13:01:22 +0100] information/ConfigItem: Instantiated 1 UserGroup.
[2021-03-27 13:01:22 +0100] information/ConfigItem: Instantiated 3 ServiceGroups.
[2021-03-27 13:01:22 +0100] information/ConfigItem: Instantiated 3 TimePeriods.
[2021-03-27 13:01:22 +0100] information/ConfigItem: Instantiated 11 Services.
[2021-03-27 13:01:22 +0100] information/ConfigItem: Instantiated 3 Zones.
[2021-03-27 13:01:22 +0100] information/ConfigItem: Instantiated 2 ScheduledDowntimes.
[2021-03-27 13:01:22 +0100] information/ConfigItem: Instantiated 2 HostGroups.
[2021-03-27 13:01:22 +0100] information/ConfigItem: Instantiated 2 NotificationCommands.
[2021-03-27 13:01:22 +0100] information/ConfigItem: Instantiated 12 Notifications.
[2021-03-27 13:01:22 +0100] information/ConfigItem: Instantiated 1 IcingaApplication.
[2021-03-27 13:01:22 +0100] information/ConfigItem: Instantiated 1 Host.
[2021-03-27 13:01:22 +0100] information/ConfigItem: Instantiated 1 Endpoint.
[2021-03-27 13:01:22 +0100] information/ConfigItem: Instantiated 235 CheckCommands.
[2021-03-27 13:01:22 +0100] information/ScriptGlobal: Dumping variables to file '/var/cache/icinga2/icinga2.vars'
[2021-03-27 13:01:22 +0100] information/ConfigObject: Restoring program state from file '/var/lib/icinga2/icinga2.state'
[2021-03-27 13:01:22 +0100] information/ConfigObject: Restored 283 objects. Loaded 28 new objects without state.
[2021-03-27 13:01:22 +0100] information/ConfigItem: Triggering Start signal for config items
[2021-03-27 13:01:22 +0100] information/NotificationComponent: 'notification' started.
[2021-03-27 13:01:22 +0100] information/CheckerComponent: 'checker' started.
[2021-03-27 13:01:22 +0100] information/ConfigItem: Activated all objects.
[2021-03-27 13:02:22 +0100] information/ConfigObjectUtility: Created and activated object '49715e0e1cec!load!142dd4c9-8442-4eb0-b002-3de46b96950e' of type 'Downtime'.
[2021-03-27 13:02:22 +0100] information/Downtime: Added downtime '49715e0e1cec!load!142dd4c9-8442-4eb0-b002-3de46b96950e' between '2021-03-28 03:00:00' and '2021-03-28 04:00:00', author: 'icingaadmin', fixed
[2021-03-27 13:02:22 +0100] information/ConfigObjectUtility: Created and activated object '49715e0e1cec!8ec241c9-efe0-46c5-a3be-6045e26ce7ed' of type 'Downtime'.
[2021-03-27 13:02:22 +0100] information/Downtime: Added downtime '49715e0e1cec!8ec241c9-efe0-46c5-a3be-6045e26ce7ed' between '2021-03-28 03:15:00' and '2021-03-28 03:45:00', author: 'me', fixed
[2021-03-27 13:02:58 +0100] information/Checkable: Checkable '49715e0e1cec!disk /' has 1 notification(s). Checking filters for type 'Problem', sends will be logged.
[2021-03-27 13:02:58 +0100] information/Notification: Sending 'Problem' notification '49715e0e1cec!disk /!mail-icingaadmin' for user 'icingaadmin'
[2021-03-27 13:02:58 +0100] information/Notification: Completed sending 'Problem' notification '49715e0e1cec!disk /!mail-icingaadmin' for checkable '49715e0e1cec!disk /' and user 'icingaadmin' using command 'mail-service-notification'.
[2021-03-27 13:03:10 +0100] information/Checkable: Checkable '49715e0e1cec!http' has 1 notification(s). Checking filters for type 'Problem', sends will be logged.
[2021-03-27 13:03:10 +0100] information/Notification: Sending 'Problem' notification '49715e0e1cec!http!mail-icingaadmin' for user 'icingaadmin'
[2021-03-27 13:03:10 +0100] information/Notification: Completed sending 'Problem' notification '49715e0e1cec!http!mail-icingaadmin' for checkable '49715e0e1cec!http' and user 'icingaadmin' using command 'mail-service-notification'.
[2021-03-27 13:03:19 +0100] information/Checkable: Checkable '49715e0e1cec!ping6' has 1 notification(s). Checking filters for type 'Problem', sends will be logged.
[2021-03-27 13:03:19 +0100] information/Notification: Sending 'Problem' notification '49715e0e1cec!ping6!mail-icingaadmin' for user 'icingaadmin'
[2021-03-27 13:03:19 +0100] information/Notification: Completed sending 'Problem' notification '49715e0e1cec!ping6!mail-icingaadmin' for checkable '49715e0e1cec!ping6' and user 'icingaadmin' using command 'mail-service-notification'.
[2021-03-27 13:03:20 +0100] information/Checkable: Checkable '49715e0e1cec!ssh' has 1 notification(s). Checking filters for type 'Problem', sends will be logged.
[2021-03-27 13:03:20 +0100] information/Notification: Sending 'Problem' notification '49715e0e1cec!ssh!mail-icingaadmin' for user 'icingaadmin'
[2021-03-27 13:03:20 +0100] information/Notification: Completed sending 'Problem' notification '49715e0e1cec!ssh!mail-icingaadmin' for checkable '49715e0e1cec!ssh' and user 'icingaadmin' using command 'mail-service-notification'.
[2021-03-27 13:03:35 +0100] information/Checkable: Checkable '49715e0e1cec!disk' has 1 notification(s). Checking filters for type 'Problem', sends will be logged.
[2021-03-27 13:03:35 +0100] information/Notification: Sending 'Problem' notification '49715e0e1cec!disk!mail-icingaadmin' for user 'icingaadmin'
[2021-03-27 13:03:35 +0100] information/Notification: Completed sending 'Problem' notification '49715e0e1cec!disk!mail-icingaadmin' for checkable '49715e0e1cec!disk' and user 'icingaadmin' using command 'mail-service-notification'.
^C[2021-01-14 13:03:58 +0100] information/Application: Received request to shut down.
[2021-01-14 13:03:58 +0100] information/Application: Received request to shut down.
[2021-03-27 13:03:58 +0100] information/Application: Shutting down...
[2021-03-27 13:03:58 +0100] information/CheckerComponent: 'checker' stopped.
[2021-03-27 13:03:58 +0100] information/NotificationComponent: 'notification' stopped.
[2021-03-27 13:03:58 +0100] information/ConfigObject: Dumping program state to file '/var/lib/icinga2/icinga2.state'
[2021-03-27 13:03:58 +0100] information/IcingaApplication: Icinga has shut down.
➜  icinga2 git:(master) find 8001 -name '*8ec241c9-efe0-46c5-a3be-6045e26ce7ed*'
8001/var/lib/icinga2/api/packages/_api/601a910f-4f1e-4911-b656-e85ea7787650/conf.d/downtimes/49715e0e1cec!8ec241c9-efe0-46c5-a3be-6045e26ce7ed.conf
➜  icinga2 git:(master) cat 8001/var/lib/icinga2/api/packages/_api/601a910f-4f1e-4911-b656-e85ea7787650/conf.d/downtimes/49715e0e1cec\!8ec241c9-efe0-46c5-a3be-6045e26ce7ed.conf
object Downtime "8ec241c9-efe0-46c5-a3be-6045e26ce7ed" ignore_on_error {
	author = "me"
	comment = " "
	config_owner = "49715e0e1cec!8001"
	duration = 0.000000
	end_time = 1616895900.000000
	entry_time = 1616846542.273560
	fixed = true
	host_name = "49715e0e1cec"
	scheduled_by = "49715e0e1cec!8001"
	start_time = 1616894100.000000
	triggered_by = ""
	version = 1616846542.273616
}
➜  icinga2 git:(master) date -r 1616894100
So 28 Mär 2021 03:15:00 CEST
➜  icinga2 git:(master) date -r 1616895900
So 28 Mär 2021 03:45:00 CEST
➜  icinga2 git:(master)
```

File: doc/09-object-types.md [hunk 1]
```diff
@@ -648,6 +648,12 @@ object ScheduledDowntime "some-downtime" {
 }
 ```
 
+Icinga 2 takes the OS' time zone including DST changes into account.
+
+Times inside DST changes are interpreted before the DST changes,
+i.e. 2020-10-25 2:30 means the CEST one
+and (the actually not existing) 2021-2-28 2:30 refers to CET (3:30 CEST).
```
Comment (julianbrost) [inline doc/09-object-types.md:655]: Please add a leading zero to `2021-02-28`.

**Review by julianbrost — CHANGES_REQUESTED** [review]
Do you think readers from other time zones will be able to follow this without additional explanation? Maybe rephrase to "on 2020-10-25, where a jump from 2:59 CEST to 2:00 CET happens, 2:30 is interpreted as CEST".

**Review by julianbrost — APPROVED** [review]
